cityofaustin/atd-vz-data
# CI workflow for applying migrations and metadata

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13294

This update adds a GH action to apply Hasura migrations and metadata when merging to `master` or `production` branches. Mostly copied from the [Moped workflow](https://github.com/cityofaustin/atd-moped/blob/main/.github/workflows/atd_moped_database.yml) but reworked since we need to get the Hasura secrets out of the `ZAPPA_SETTINGS` secret. I went with [jq](https://jqlang.github.io/jq/) to handle that since we've used it in the past.

Having the general shape of `ZAPPA_SETTINGS` could be helpful for reviewing what the bash script is doing.
```json
{
    "production": {
        "aws_environment_variables": {
            "HASURA_ENDPOINT": "https://vzd.austinmobility.io/v1/graphql",
            "HASURA_ADMIN_SECRET": "thisisaprodsecret"
        },
    },
    "staging": {
        "aws_environment_variables": {
            "HASURA_ENDPOINT": "https://vzd-staging.austinmobility.io/v1/graphql",
            "HASURA_ADMIN_SECRET": "thisisastagingsecret"
        },
    }
}
```

## Testing
**URL to test:** 

**Steps to test:**
1. Read the code and updated readme to see if it makes sense. 🙏
2. Double check the flags on the CLI commands that intend to keep any user interactions from needing to happen?
- [hasura migrate apply](https://hasura.io/docs/latest/hasura-cli/commands/hasura_migrate_apply/)
- [hasura metadata apply](https://hasura.io/docs/latest/hasura-cli/commands/hasura_metadata_apply/)
3. Other than that, this will be another one that we can merge, test, and patch if needed.
---
#### Ship list
